src/service/routing/spec.clj

```diff
@@ -50,13 +60,20 @@
     (GET "/direction/:coordinates" []
       :summary "direction with clojure.spec"
       :path-params [coordinates :- ::coordinate-regex]
-      :query-params [{steps :- boolean? false}]
+      :query-params [{steps :- boolean? false}
+                     {radiuses :- ::radiuses-regex nil}
+                     {alternatives :- boolean? false}
+                     {language :- string? "en"}]
       :return ::direction
       (ok (let [coords (map zipmap (repeat [:lon :lat])
-                                   (->coordinates coordinates))]
+                                   (->coordinates coordinates))
+                rads (if (nil? radiuses) nil (->radiuses radiuses))]
```

Comment:
small style issue but prefer to use `when` if one of the return values is `nil` ;)